thrift-logger/src/main/java/com/pinterest/singer/client/logback/AuditableLogbackThriftLoggerFactory.java

```diff
@@ -0,0 +1,169 @@
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.pinterest.singer.client.logback;
+
+import com.pinterest.singer.client.AuditableLogbackThriftLogger;
+import com.pinterest.singer.client.ThriftLogger;
+import com.pinterest.singer.client.ThriftLoggerConfig;
+import com.pinterest.singer.loggingaudit.client.LoggingAuditClient;
+import com.pinterest.singer.utils.CommonUtils;
+import com.pinterest.singer.loggingaudit.client.utils.ConfigUtils;
+import com.pinterest.singer.loggingaudit.thrift.LoggingAuditStage;
+import com.pinterest.singer.loggingaudit.thrift.configuration.LoggingAuditClientConfig;
+import com.pinterest.singer.loggingaudit.thrift.configuration.AuditConfig;
+import com.pinterest.singer.metrics.OpenTsdbMetricConverter;
+import com.pinterest.singer.thrift.LogMessage;
+
+import ch.qos.logback.core.Appender;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.commons.configuration.SubsetConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Factory that creates a AuditableLogbackThriftLogger.
+ *
+ * By default we use a file-rolling appender.
+ */
+public class AuditableLogbackThriftLoggerFactory extends LogbackThriftLoggerFactory{
+
+  private static Logger LOG = LoggerFactory.getLogger(AuditableLogbackThriftLoggerFactory.class);
+
+  private static final String AUDIT_THRIFT_LOGGER_CLIENT_INIT_EXCEPTION = "audit.thrift_logger.audit_client.init.exception";
+  private static LoggingAuditClient loggingAuditClient;
+
+  @Deprecated
+  public AuditableLogbackThriftLoggerFactory(File basePath, int rotateThresholdKBytes) {
+    super(basePath, rotateThresholdKBytes);
+  }
+
+  public AuditableLogbackThriftLoggerFactory() {
+    super();
+  }
+
+  public static void createLoggingAuditClientFromConfigFile(String filePath){
+    if (loggingAuditClient != null){
+      return;
+    }
+    try {
+      LoggingAuditClientConfig config = ConfigUtils.parseFileBasedLoggingAuditClientConfig(filePath);
+      loggingAuditClient = new LoggingAuditClient(config);
+    } catch (ConfigurationException e){
+      loggingAuditClient = null;
+      LOG.error("Couldn't successfully create LoggingAuditClient and thus LoggingAuditEvent can not be sent out.");
+      OpenTsdbMetricConverter.incr(AUDIT_THRIFT_LOGGER_CLIENT_INIT_EXCEPTION , "stage=" + LoggingAuditStage.THRIFTLOGGER.toString(), "host=" + CommonUtils.getHostName());
+    }
+  }
+
+  public static void createLoggingAuditClientFromKVs(Map<String, String> properties){
```

Comment:
this method is not used anywhere. do we need to keep it?

Reply:
I will update the name of this method to be createLoggingAuditClientFromMap to be consistent with the name in the LoggingAuditClient.java.

I think we need to keep it.

It is not used in the current code but may be used by any applications that import the jars.   If adding an extra config file about audit related properties is an overhead for the application,  it can just pass in a hashmap to create logging audit client.